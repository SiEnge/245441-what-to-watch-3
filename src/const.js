export const MINUTE_IN_ONE_HOUR = 60;
export const TEN_MINUTES = 10;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

export const AppRoute = {
  LOGIN: `/login`,
  REVIEW: `/review`,
  MY_LIST: `/mylist`,
  FILMS: `/films`,
  PLAYER: `/player`,
  ROOT: `/`,
};

export const TabsType = {
  OVERVIEW: `Overview`,
  DETAIL: `Details`,
  REVIEWS: `Reviews`
};

export const ScoreType = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};

// const GenreType = {
//   ALL_GENRES: `All genres`,
//   COMEDIAS: `Comedies`,
//   CRIME: `Crime`,
//   DOCUMENTARY: `Documentary`,
//   DRAMAS: `Dramas`,
//   HORROR: `Horror`,
//   ROMANCE: `Romance`,
//   SCI_FI: `Sci-Fi`,
//   THRILLERS: `Thrillers`,
// };
