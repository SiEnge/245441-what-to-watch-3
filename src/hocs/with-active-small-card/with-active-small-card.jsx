import React, {PureComponent} from "react";
const VIDEO_DELAY = 1000;

export const withActiveSmallCard = (Component) => {
  return class WithActiveSmallCard extends PureComponent {
    constructor(props) {
      super(props);

      this.state = {
        activeSmallCard: -1,
      };

      this._handleCardHover = this._handleCardHover.bind(this);
    }

    _handleCardHover(id) {
      if (id !== this.state.activeSmallCard) {
        if (id === -1) {
          this.setState({
            activeSmallCard: id,
          });
        } else {
          setTimeout(() => {
            this.setState({
              activeSmallCard: id,
            });
          }, VIDEO_DELAY);
        }
      }
    }

    render() {
      return (
        <Component {...this.props} activeSmallCard={this.state.activeSmallCard} onSmallCardHover={this._handleCardHover} />
      );
    }
  };
};
