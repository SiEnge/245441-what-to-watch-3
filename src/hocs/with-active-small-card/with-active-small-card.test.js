import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import {withActiveSmallCard} from "./with-active-small-card.jsx";
import {noop} from "../../const.js";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
};

const MockComponentWrapped = withActiveSmallCard(MockComponent);

it(`withActiveSmallCard is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      activeSmallCard={1}
      onSmallCardHover={noop}
    />
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
