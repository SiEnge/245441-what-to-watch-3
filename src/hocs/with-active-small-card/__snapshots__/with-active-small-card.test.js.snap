// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withActiveSmallCard is rendered correctly 1`] = `<div />`;
